pkgs/build-support/bintools-wrapper/setup-hook.sh

```diff
@@ -24,7 +24,8 @@ bintoolsWrapper_addLDVars () {
         # Python and Haskell packages often only have directories like $out/lib/ghc-8.4.3/ or
         # $out/lib/python3.6/, so having them in LDFLAGS just makes the linker search unnecessary
         # directories and bloats the size of the environment variable space.
-        if [[ -n "$(echo $1/lib/lib*)" ]]; then
+        local -a glob=( $1/lib/lib* )
+        if [ "${#glob[*]}" -gt 0 ]; then
```

Comment:
Why go to single `[[`?

Reply:
Just my personal preference. In this case both can be used.

Reply:
Since we are not trying to be POSIX compatible I think it's best to always use `[[` `]]` and pretend that `[` `]` don't exist.

Reply:
For example here you can use `[[ "${#glob[*]}" > 0 ]]`. That syntax also reacts better with missing quotes. Eg:
```
foo="x -o true"
[ x = $foo ] # => 0
[[ x = $foo ]] # => 1
```